src/components/LessonHighlights/LessonHighlights.tsx

```diff
@@ -38,27 +39,40 @@ const LessonHighlights: FC<Props> = ({ language }) => {
   const { t } = useTranslation()
   const lessons = language === "fi" ? lessonsFI : lessonsEN
 
+  const { data: bookmarkedData } = useGetAllBookmarks(lessons)
+
+  let bookmarked
+
   return (
     <Container>
       <H2>{t("COACHING.LESSONS")}</H2>
       <P>{t("COACHING.LESSONS_TEXT")}</P>
 
       <Lessons>
-        {lessons.nodes.map((lesson: ContentfulLesson) => (
-          <LessonCard
-            slug={`${lesson?.slug}`}
-            name={lesson?.lessonName}
-            key={lesson?.slug as string}
-            bookmarked={false}
-            onClick={() => {}}
-            loading={false}
-            path={`/lesson/${lesson?.slug}`}
-            lesson={lesson}
-            readingTime={lesson?.lessonContent?.fields?.readingTime?.minutes}
-            cover={lesson?.cover?.fluid as FluidObject}
-            excerpt={lesson?.lessonContent?.fields?.excerpt}
-          />
-        ))}
+        {lessons.nodes.map((lesson: ContentfulLesson) => {
+          const bookmarkSearch = bookmarkedData?.find(
+            (item: any) => item.slug == lesson.slug
+          )
+
+          bookmarkSearch === undefined
+            ? (bookmarked = false)
+            : (bookmarked = true)
```

Comment:
Same as with the comment of weeks. Also you have big closure error here with bookmarked being a variable outside of the loop. That means that every compont always reference that same variable and it will also have the value from the last item in the array (because it's reassigned on every new loop)